client/src/main/java/org/asynchttpclient/netty/channel/DefaultChannelPool.java

```diff
@@ -257,7 +252,7 @@ private boolean offer0(Channel channel, Object partitionKey, long now) {
     }
 
     private void registerChannelCreation(Channel channel, Object partitionKey, long now) {
-        if (channelId2Creation.containsKey(partitionKey)) {
+        if (partitions.containsKey(partitionKey)) {
```

Comment:
Yeah, right (was looking at old code and got confused).
So yes. And please compute channelId only once.